codec-http/src/test/java/io/netty/handler/codec/http/websocketx/WebSocketHandshakeHandOverTest.java

```diff
@@ -95,6 +97,85 @@ protected void channelRead0(ChannelHandlerContext ctx, Object msg) throws Except
+        EmbeddedChannel serverChannel = createServerChannel(new SimpleChannelInboundHandler<Object>() {
+            @Override
+            public void userEventTriggered(ChannelHandlerContext ctx, Object evt) {
+                if (evt == ServerHandshakeStateEvent.HANDSHAKE_COMPLETE) {
+                    serverReceivedHandshake = true;
+                    // immediately send a message to the client on connect
+                    ctx.writeAndFlush(new TextWebSocketFrame("abc"));
+                } else if (evt instanceof WebSocketServerProtocolHandler.HandshakeComplete) {
+                    serverHandshakeComplete = (WebSocketServerProtocolHandler.HandshakeComplete) evt;
+                }
+            }
+
+            @Override
+            protected void channelRead0(ChannelHandlerContext ctx, Object msg) throws Exception {
+            }
+        });
+
+        EmbeddedChannel clientChannel = createClientChannel(new SimpleChannelInboundHandler<Object>() {
+            @Override
+            public void userEventTriggered(ChannelHandlerContext ctx, Object evt) {
+                if (evt == ClientHandshakeStateEvent.HANDSHAKE_COMPLETE) {
+                    clientReceivedHandshake = true;
+                }
+
+                if (evt == ClientHandshakeStateEvent.HANDSHAKE_TIMEOUT) {
+                    clientHandshakeTimeout = true;
+                }
+            }
+
+            @Override
+            protected void channelRead0(ChannelHandlerContext ctx, Object msg) throws Exception {
+                if (msg instanceof TextWebSocketFrame) {
+                    clientReceivedMessage = true;
+                }
+            }
+        }, 100);
+        // Client send the handshake request to server
+        transferAllDataWithMerge(clientChannel, serverChannel);
+        // Server do not send the response back
+        // transferAllDataWithMerge(serverChannel, clientChannel);
+        WebSocketClientProtocolHandshakeHandler handshakeHandler =
+                (WebSocketClientProtocolHandshakeHandler) clientChannel
+                        .pipeline().get(WebSocketClientProtocolHandshakeHandler.class.getName());
+
+        while (!handshakeHandler.getHandshakeFuture().isDone()) {
+            Thread.sleep(10);
+            // We need to run all pending tasks as the handshake timeout is scheduled on the EventLoop.
+            clientChannel.runScheduledPendingTasks();
+        }
+        assertTrue(clientHandshakeTimeout);
+        assertFalse(clientReceivedHandshake);
+        assertFalse(clientReceivedMessage);
+        // Should throw WebSocketHandshakeException
+        try {
+            handshakeHandler.getHandshakeFuture().syncUninterruptibly();
+        } finally {
+            cleanChannelData(serverChannel);
+        }
+    }
+
+    /**
+     * Clean all pending data in the channel.
+     * @param channel The channel to be cleaned up
+     */
+    private static void cleanChannelData(EmbeddedChannel channel) {
```

Comment:
remove this method and just use `channel.finishAndReleaseAll()` above.

Reply:
Done